### DISCUSSION TITLE
How to use the extension

### DISCUSSION THREAD

**Discussion by AhmedTawfek** [dotnet/DataGridExtensions#88] [Q&A]
My question may appear silly, but i couldn't manage how to use this extension ...
I already downloaded the extension from nuGet, added the namespace at the window.xaml and enabled the filter to the datagrid, now how can i use the filter as popUpMenu ? when i used this line "dgx:DataGridFilterColumn.Template="{StaticResource FilterWithPopup}", it gives me an error because it can't find the resource.
Am i missing something? sorry am new to WPF.

**Comment by tom-englert** [comment]
You can find samples for the main use cases in the sample app 
https://github.com/dotnet/DataGridExtensions/tree/master/src%2FDataGridExtensionsSample

**Reply by AhmedTawfek** [reply]
Thanks for your great effort and work.
I managed to use the extension successfully and everything is going right.
Now my problem is i want to make column splitter like in this one : [stackoverflow](https://stackoverflow.com/questions/12970463/split-the-column-of-datagrid-in-wpf)

And i have enabled the filter but it's not showing up like other columns : [screenshot of the datagrid](http://epvpimg.com/MnWPbab)
The filter doesn't work when using this custom style, can you help me please? thanks in advance.

**Comment by tom-englert** [comment]
I think you must ensure you're new column styles use the correct template:

https://github.com/dotnet/DataGridExtensions/blob/master/src/DataGridExtensions/Themes/Generic.xaml#L182-L193

**Reply by AhmedTawfek** [reply]
I don't know if this is a bug, but what i want is when the user edits a cell, the cell background color changes.
The problem is the cell background color is changed, but after some seconds it goes back to the default color again.
And if i don't use the filter extension in the datagrid, the cell background color is changed forever as i want.

[gif of what is happening](https://imgur.com/a/50O5qwe)

**Reply by tom-englert** [reply]
DGX is introducing additional styles, so make sure to change the right ones

**Reply by AhmedTawfek** [reply]
Sorry, but i didn't get it.
Can you tell me more please about what i have to use ?

**Reply by tom-englert** [reply]
It was just a quick guess, as I don't know what you are doing. 
However don't expect too much here, open source does not imply free of charge tutorials or contact programming.
